### PR TITLE
CPS-601: Add Prospects Scenarios

### PR DESCRIPTION
**Pull request by lisandro-compucorp**
## Overview
This PR adds the testing scenarios related to prospects extension, along with the data setup needed.

## Before
Neither the scenarios nor the data existed.

## After
The new scenarios were added:
- Menu displayed on a prospect:
Civicase51_Prospects_-_Case_Overview_-_Menu_Displayed_0_document_0_desktop
- Click on Convert to Contribution:
Civicase51_Prospects_-_Case_Overview_-_Convert_to_Contribution_0_document_0_desktop
- Click on Convert to Pledge:
Civicase51_Prospects_-_Case_Overview_-_Convert_to_Pledge_0_document_0_desktop
Also, it was added a prospect in the data setup.

## Comments
- The click on menu items was refactored also on manage cases scenarios. Now it is selected by the text of the link, not for the position. It was tested locally that the change worked as expected.
- The file `action-dropdown-email-case-manager.js` was deleted since it was not in use anymore.

### CODE REVIEW

File: engine_scripts/puppet/manage-prospectings/click-contribution-button.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+const Utility = require('../utility.js');
+
+module.exports = async (page, scenario, vp) => {
+  const utility = new Utility(page, scenario, vp);
+  await utility.waitForAngular();
+  await utility.waitForLoadingComplete();
+
+  await page.click('.civicase__case-header__action-icon.btn.btn-primary');
```
Comment (deb1990) [inline engine_scripts/puppet/manage-prospectings/click-contribution-button.js]: These are not required, you can just call 
`await require('./action-dropdown')(page, scenario, vp);`

File: engine_scripts/puppet/manage-prospectings/click-pledge-button.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+const Utility = require('../utility.js');
+
+module.exports = async (page, scenario, vp) => {
+  const utility = new Utility(page, scenario, vp);
+  await utility.waitForAngular();
+  await utility.waitForLoadingComplete();
+
+  await page.click('.civicase__case-header__action-icon');
```
Comment (deb1990) [inline engine_scripts/puppet/manage-prospectings/click-pledge-button.js]: These are not required, you can just call 
`await require('./action-dropdown')(page, scenario, vp);`

File: gulp-tasks/data-setup-steps/activity.service.js [hunk 3]
```diff
@@ -96,4 +97,17 @@ function createAwardsActivities () {
   });
 }
 
+/**
+ * Create Prospecting Activities
+ */
+function createProspectingActivities () {
```
Comment (deb1990) [inline gulp-tasks/data-setup-steps/activity.service.js]: I dont think we need to create an activity for prospect? If not please remove.

File: gulp-tasks/utils/get-active-prospecting-id.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+const cvApi = require('./cv-api.js');
+
+module.exports = getActiveProspectingId;
+
+/**
+ * Returns the ID of a application that is active and has an activity for the current
```
Comment (deb1990) [inline gulp-tasks/utils/get-active-prospecting-id.js]: Description needs to be updated.

File: scenarios/prospect/manage-prospectings.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{
+  "scenarios": [
+    {
+      "label": "Prospects - Case Overview - Menu Displayed",
```
Comment (deb1990) [inline scenarios/prospect/manage-prospectings.json]: Should be named `Manage Prospects - Case Overview - Case Action Menu Displayed` or `Manage Prospects - Case Overview - Case Action Menu`.

File: scenarios/prospect/manage-prospectings.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+      "url": "{url}/civicrm/case/a/?case_type_category={ctc-prospecting}#/case/list?cf=%7B\"case_type_category\":\"{ctc-prospecting}\"%7D&caseId={prospectingId}",
+      "waitForLoadingComplete": ".civicase__summary-tab",
+      "clickSelector": ".civicase__case-header__action-icon"
+    },
+    {
+      "label": "Prospects - Case Overview - Convert to Pledge",
```
Comment (deb1990) [inline scenarios/prospect/manage-prospectings.json]: `Manage Prospects - Case Overview - Action Dropdown - <>`

File: scenarios/prospect/manage-prospectings.json [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+      "url": "{url}/civicrm/case/a/?case_type_category={ctc-prospecting}#/case/list?cf=%7B\"case_type_category\":\"{ctc-prospecting}\"%7D&caseId={prospectingId}",
+      "waitForLoadingComplete": ".civicase__summary-tab",
+      "onReadyScript": "manage-prospectings/click-pledge-button.js"
+    },
+    {
+      "label": "Prospects - Case Overview - Convert to Contribution",
```
Comment (deb1990) [inline scenarios/prospect/manage-prospectings.json]: `Manage Prospects - Case Overview - Action Dropdown - <>`

**Review by deb1990 — CHANGES_REQUESTED** [review]

File: engine_scripts/puppet/manage-prospectings/click-contribution-button.js (hunk 1 above)
Comment (lisandro-compucorp) [reply engine_scripts/puppet/manage-prospectings/click-contribution-button.js]: Done: https://github.com/compucorp/cases-product-suite-backstopjs/pull/4/commits/208836b766d97a6b0a7e7ebc39f77ab5a6fe76f9 I moved the script out of the previous folder for reuse it on both scenarios (cases and prospecting)

File: gulp-tasks/data-setup-steps/activity.service.js (hunk 3 above)
Comment (lisandro-compucorp) [reply gulp-tasks/data-setup-steps/activity.service.js]: Removed: https://github.com/compucorp/cases-product-suite-backstopjs/pull/4/commits/fc9a75fb6671aa8900bc018efa1574258e33659c In consequence I replaced the way to load the prospect Id for replacing in the URL

File: scenarios/prospect/manage-prospectings.json (hunk 5 above)
Comment (lisandro-compucorp) [reply scenarios/prospect/manage-prospectings.json]: Done: https://github.com/compucorp/cases-product-suite-backstopjs/pull/4/commits/37e77133f29f04c253daacfe0a0b0f6c29b54d62

File: gulp-tasks/utils/get-active-prospecting-id.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    sequential: 1,
+    is_deleted: 0,
+    case_type_id: prospectTypeId.values[0].id,
+    return: ['id'],
+    options: { limit: 1 }
+  });
+
+  if (!prospecting.count) {
+    throw new Error('Please add a prospecting');
+  }
```
Comment (deb1990) [inline gulp-tasks/utils/get-active-prospecting-id.js:33]: This check is not necessary, as you are creating the prospect yourself.

**Review by deb1990 — CHANGES_REQUESTED** [review]

**Review by deb1990 — APPROVED** [review]

**Pull request merged**